substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/pecoff/cv/CVSymbolSectionImpl.java

```diff
@@ -186,4 +186,27 @@ int getCurrentOffset() {
     void addRecord(CVSymbolRecord record) {
         cvRecords.add(record);
     }
+
+    /**
+     * Mark an offset:segment relocation site for linker or loader fixup.
+     *
+     * @param buffer output buffer
+     * @param initialPos position of fixup in output buffer
+     * @param offset offset to add to the fixup
+     * @param symbolName symbolname to reference
+     * @return new position in output buffer
+     */
+    public int markRelocationSite(byte[] buffer, int initialPos, String symbolName, Long offset) {
+        int pos = initialPos;
+        if (buffer != null) {
+            markRelocationSite(pos, ObjectFile.RelocationKind.SECREL_4, symbolName, offset);
+            pos += ObjectFile.RelocationKind.getRelocationSize(ObjectFile.RelocationKind.SECREL_4);
+            markRelocationSite(pos, ObjectFile.RelocationKind.SECTION_2, symbolName, 0);
+            pos += ObjectFile.RelocationKind.getRelocationSize(ObjectFile.RelocationKind.SECTION_2);
+        } else {
+            pos = CVUtil.putInt(0, buffer, pos);
+            pos = CVUtil.putShort((short) 0, buffer, pos);
+        }
+        return pos;
+    }
```

Comment:
I'd remove the `offset` parameter because it seems that we don't need it for now (we always pass `0`).

We could also simplify the implementation a bit using a helper method like this:
```suggestion
    public int markRelocationSite(byte[] buffer, int initialPos, String symbolName) {
        int pos = initialPos;
        pos += markRelocationSite(buffer, pos, symbolName, RelocationKind.SECREL_4);
        pos += markRelocationSite(buffer, pos, symbolName, RelocationKind.SECTION_2);
        return pos;
    }

    private int markRelocationSite(byte[] buffer, int initialPos, String symbolName, RelocationKind k) {
        if (buffer != null) {
            markRelocationSite(initialPos, k, symbolName, 0);
        }
        return initialPos + RelocationKind.getRelocationSize(k);
    }
```

Reply:
Done, though personally my taste is to keep the offset param.

Reply:
@pejovica the use case for the offset parameter is when defining static member objects; they are defined as an offset from svm_heap.  I will keep the non-offset API (as it is the most common case) and add an API that has an offset parameter.